sklearn/utils/validation.py

```diff
@@ -8,35 +8,18 @@
 # License: BSD 3 clause
 import warnings
 import numbers
+from inspect import getargspec
 
 import numpy as np
 import scipy.sparse as sp
 
 from ..externals import six
 from ..utils.fixes import signature
+from ..exceptions import (DataConversionWarning, NonBLASDotWarning,
+                          NotFittedError)
```

Comment:
You may want to use the `deprecated` decorator on each one of these warnings though (as you did [here](https://github.com/scikit-learn/scikit-learn/pull/4826/files#diff-e1790731a970adbbeecfaad894810227R26) )